WordPress/gutenberg
# Refactor the HierarchicalTermSelector so that it does not cause unnecessary loading of terms.

## Description

Related: #10169, #16268

* Refactor the HierarchicalTermSelector so that it does not cause unnecessary loading of terms. Use the data from the first set of terms we get.
* Changed from class components to functional components.



## How has this been tested?
1. Go edit post.
2. Close and open the panel for categories.
3. Check request for wp-api.

## Screenshots 

## Types of changes

Bug fix



## Checklist:
